src/ddsrt/src/string.c

```diff
@@ -219,3 +219,41 @@ ddsrt_strndup(const char *str, size_t len)
 
   return s;
 }
+
+void
+ddsrt_str_trim(char *str)
+{
+  int index, i;
+
+  if (str == NULL) {
+    return;
+  }
+  
+  /* trim leading white spaces */
+  index = 0;
+  while(str[index] == ' ' || str[index] == '\t' || str[index] == '\n') {
+    index++;
+  }
+
+  /* shift all trailing characters to its left */
+  i = 0;
+  while(str[i + index] != '\0') {
+    str[i] = str[i + index];
+    i++;
+  }
+  str[i] = '\0'; /* terminate string with NULL */
```

Comment:
For the way it is used, it could also be changed to only trim the right-hand white space and return a pointer to the first non-white space character, eliminating the shifting. That's probably what I would have done, but this interface is probably less likely to be used incorrectly 🙂

What I don't really like is doing the shifting even over a distance of 0, which I suspect is the common case. That's easily remedied, of course.

```suggestion
  if (index > 0) {
    i = 0;
    while(str[i + index] != '\0') {
      str[i] = str[i + index];
      i++;
    }
    str[i] = '\0'; /* terminate string with NULL */
  }
```